TwoFactorAuth/Test/Unit/Model/Config/Backend/Duo/ApiHostnameTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+declare(strict_types=1);
+
+namespace Magento\TwoFactorAuth\Test\Unit\Model\Config\Backend;
+
+use Magento\Framework\Exception\ValidatorException;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\TwoFactorAuth\Model\Config\Backend\Duo\ApiHostname;
+use PHPUnit\Framework\TestCase;
+
+class ApiHostnameTest extends TestCase
+{
+    /**
+     * @var ApiHostname
+     */
+    private $model;
+
+    /**
+     * @inheritDoc
+     */
+    protected function setUp()
+    {
+        $objectManager = new ObjectManager($this);
+        $this->model = $objectManager->getObject(ApiHostname::class);
+    }
+
+    /**
+     * @dataProvider valuesDataProvider
+     */
+    public function testBefore($value, $isValid): void
+    {
+        if (!$isValid) {
```

Comment:
I'm not sure what you mean. The dataprovider has all the tested cases including things like `foo`. If the model already contained an invalid value and they try to save it again it will throw an error due to this validation.

Reply:
Sorry for confusion - I was not talking about this test in particular. I was talking about the case when a merchant had already 2FA module installed and then upgrades it - what would happen if they have an invalid value configured?

Reply:
Currently nothing. If you think it's important we could add an update script that validates it and clears it if it's bad. It shouldn't work anyway if it's bad so I don't see the harm.

Reply:
That's alright, we can leave it as is